workers/tests/integrationtests/test_oqt.py

```diff
@@ -14,157 +20,114 @@
 class TestOqt(unittest.TestCase):
     def setUp(self):
         # Heidelberg
+        self.name = "GhsPopComparisonBuildings"
+        self.report_name = "SimpleReport"
+        self.layer_name = "building_count"
         self.dataset = "regions"
         self.feature_id = "3"
         self.fid_field = "ogc_fid"
         self.feature = asyncio.run(
             db_client.get_feature_from_db(self.dataset, feature_id=self.feature_id)
         )
 
-    @oqt_vcr.use_cassette()
-    def test_create_indicator_feature(self):
-        """Test creating indicator from scratch."""
-        # From scratch
-        indicator = asyncio.run(
-            oqt.create_indicator(
-                "GhsPopComparisonBuildings", "building_count", feature=self.feature
-            )
-        )
+    def run_tests(self, indicator):
         self.assertIsNotNone(indicator.result.label)
         self.assertIsNotNone(indicator.result.value)
         self.assertIsNotNone(indicator.result.description)
         self.assertIsNotNone(indicator.result.svg)
 
     @oqt_vcr.use_cassette()
-    def test_create_indicator_invalid_dataset(self):
-        with self.assertRaises(ValueError):
-            asyncio.run(
-                oqt.create_indicator(
-                    "GhsPopComparisonBuildings",
-                    "building_count",
-                    dataset="foo",
-                    feature_id=self.feature_id,
-                )
-            )
-
-    @oqt_vcr.use_cassette()
-    def test_create_indicator_dataset_invalid_fid_field(self):
-        with self.assertRaises(ValueError):
-            asyncio.run(
-                oqt.create_indicator(
-                    "GhsPopComparisonBuildings",
-                    "building_count",
-                    dataset=self.dataset,
-                    feature_id=self.feature_id,
-                    fid_field="foo",
-                )
-            )
+    def test_create_indicator_bpolys(self):
+        """Test creating indicator from scratch."""
+        param = IndicatorBpolys(
+            name=self.name,
+            layerName=self.layer_name,
+            bpolys=self.feature,
+        )
+        indicator = asyncio.run(oqt.create_indicator(param))
+        self.run_tests(indicator)
 
```

Comment:
Why is this test not necessary anymore? Through which functionality or test is this case covered now?

Reply:
Because validation of parameters, are done during the instantiation of the pydantic data classes. Those are tested in `test_api_request_models.py`. Since in this PR the `oqt.create_indicator` function takes a pydantic data class as input, this test is obsolete.